### DISCUSSION TITLE
DefaultConfig of useSWR should not be overwritten with undefined

### DISCUSSION THREAD

**Discussion by kazukinagata** [vercel/swr#2570] [Ideas]
We have a custom hook called useUsers that wraps useSWR as follows. This hook accepts the revalidateIfStale SWR option, with a default value of undefined.

```javascript
function useUsers({
  revalidateIfStale
}) {
  const key = `path_to_the_users`;
  return useSWR(key, fetcher, {revalidateIfStale});
}
```

I expected that `defaultConfig.revalidateIfStale /** this is true **/` would be applied when I passed an empty object to useUsers. However, the config.revalidateIfStale was merged with mergeObjects as undefined, which was contrary to my expectations.

You can find the code here:
https://github.com/vercel/swr/blob/342b82c818454b05c05ae7294c531674926539db/_internal/utils/helper.ts#L22

Is this the intended behavior?

I would like to suggest the following implementation to avoid overwriting defaultConfig when the user passes undefined.

```javascript
const mergeObjects = (a, b) => {
  const merged = { ...a };
  for (const key in b) {
    if (typeof b[key] !== "undefined") {
      merged[key] = b[key];
    }
  }
  return merged;
}
```

**Comment by devjiwonchoi** [comment]
@kazukinagata What is the expected output for `revalidateIfStale` param passed to `useUsers`?
